src/control/lib/netdetect/netdetect.go

```diff
@@ -1207,3 +1216,21 @@ func ScanFabric(provider string) ([]FabricScan, error) {
 	}
 	return ScanResults, nil
 }
+
+// GetDeviceClass determines the device type according to what's stored in the filesystem
+// Returns an integer value corresponding to its ARP protocol hardware identifier
+// found here: https://elixir.free-electrons.com/linux/v4.0/source/include/uapi/linux/if_arp.h#L29
+func GetDeviceClass(netdev string) (int32, error) {
+	devClass, err := ioutil.ReadFile(fmt.Sprintf("/sys/class/net/%s/type", netdev))
+	if err != nil {
+		return 0, err
+	}
+
+	// minimally one byte of data and one newline character.
+	if len(devClass) < 2 {
+		return 0, errors.Errorf("/sys/class/net/%s/type has an unexpected format: %+v", netdev, devClass)
+	}
+
+	res, err := strconv.Atoi(string(devClass[:len(devClass)-1]))
```

Comment:
```go
strconv.Atoi(string(devClass))
```
Doesn't work?

Reply:
I'll double check.  I thought it had a heart attack on the trailing \n in the bytes read.